proposals/2024/10-01_projektor_integration.md

```diff
@@ -0,0 +1,232 @@
+4. **`code_coverage_run`**: Aggregates coverage data for a specific run.
+5. **`performance_results`**: Contains performance metrics, such as request count, average time, and maximum response time.
+6. **`code_quality_report`**: Stores code quality report data, including file and group names.
+7. **`test_run`**: Represents a single test run, storing counts of passed, failed, and skipped tests along with timing details.
+8. **`results_metadata`**: Contains metadata related to each test run, such as CI information and group labels.
+9. **`test_run_system_attributes`**: Stores system attributes for test runs, such as pinned status.
+10. **`test_suite`**: Represents high-level test groupings, like test suites, with success and failure counts.
+11. **`test_suite_group`**: Groups related test suites together for organized reporting.
+12. **`test_case`**: Stores individual test cases, including their results, duration, and logs.
+13. **`test_failure`**: Captures detailed information on test failures, including failure messages and stack traces.
+14. **`test_run_attachment`**: Manages attachments for a test run, like log files and screenshots.
+15. **`results_processing`**: Logs the status and errors (if any) from test results processing.
+16. **`results_processing_failure`**: Tracks specific failure cases encountered during test processing.
+17. **`shedlock`**: Used for distributed locking to prevent concurrent processing issues.
+
+Tables related to git metadata are omitted as Vela already has a robust git integration system.
+
+
+#### Proposed API Endpoints
+Vela’s backend will expose new API endpoints to support test reporting and analytics features.
+1. **`/api/v1/test-report`**: POST endpoint to submit test results for processing and storage.
+2. **`/api/v1/test-report/{run_id}`**: GET endpoint to retrieve test results for a specific test run.
+3. **`/api/v1/code-coverage`**: POST endpoint to submit code coverage data for processing and storage.
+4. **`/api/v1/code-coverage/{run_id}`**: GET endpoint to fetch code coverage data for a specific run.
+5. **`/api/v1/performance-metrics`**: POST endpoint to submit performance metrics for processing and storage.
+6. **`/api/v1/performance-metrics/{run_id}`**: GET endpoint to retrieve performance metrics for a specific run.
+7. **`/api/v1/code-quality`**: POST endpoint to submit code quality reports for processing and storage.
+8. **`/api/v1/code-quality/{run_id}`**: GET endpoint to fetch code quality data for a specific run.
+9. **`/api/v1/flaky-tests`**: GET endpoint to list flaky tests detected in the system.
+10. **`/api/v1/test-notifications`**: POST endpoint to send test result notifications to Slack channels.
+
+
+### 4. Object Store Integration
+To effectively manage test artifacts and large volumes of test result data, Go-Vela will incorporate an object store as part of the test reporting solution.
+#### Key Features of the Object Store
```

Comment:
not sure we need to list out features of object store? what about initial support for which object stores will be supported. presumably, the first integration will be S3 compatible object storage. 

then, maybe list out what the integration looks like? just add some env vars (like secrets and url) to configure? will there be admin UI/API to control more nuanced aspects like lifecycle/expiration and/or buckets?